terraform/modules/terraform-null-resources/main.tf

```diff
@@ -0,0 +1,47 @@
+
+# create a single resource
+# but only if var.module_enabled is true
+# and var.string is not empty
+resource "null_resource" "single" {
+  count = var.module_enabled && length(var.string) > 0 ? 1 : 0
+
+  triggers = {
+    value = jsonencode(var.string)
+  }
+
+  depends_on = [
+    var.module_depends_on
+  ]
+}
+
+# create one resource for every element in var.list_of_strings
+# but only if var.module_enabled is true
+# resources will be created as a list of objects
+# if the var.list_of_strings is empty no resources will be created
```

Comment:
multiline comments should follow our standard and represent whole sentences

Reply:
fixed